test/discord.test.ts

```diff
@@ -293,161 +294,407 @@ describe('calling getUserId', () => {
-            channel: {
-                send: jest.fn(async () => sentMessage),
-                startTyping: jest.fn(() => Promise.resolve()),
-                stopTyping: jest.fn(() => Promise.resolve()),
-                type: 'text',
-                name: channelName
-            }
-        };
-
-        client.emit('message', message);
+    describe('when message is "!tasks in <list>"', () => {
+        it('should call commandController.taskInListForProject with list from '
+        +   'command and project ID from channel', done => {
+            const client = setupClient();
+
+            const taskList = 'Selected for Development';
+            const projectId = 0;
+            const returnedTasks = new RichEmbed();
+
+            const commandControllerMock = new CommandControllerMockBuilder()
+                .withTasksInListForProject(jest.fn(() => Promise.resolve(returnedTasks)))
+                .build();
+
+            const mappingControllerMock = new MappingControllerMockBuilder()
+                .withGetProjectId(jest.fn().mockReturnValue(projectId))
+                .build();
+
+            const discordController = new DiscordControllerBuilder()
+                .withClient(client)
+                .withMappingController(mappingControllerMock)
+                .withCommandController(commandControllerMock)
+                .build();
+
+            const message = new MessageBuilder()
+                .withContent(`!tasks in ${taskList}`)
+                .withSend(jest.fn(async value => {
+                    if (value !== returnedTasks) {
+                        return;
+                    }
+
+                    expect(commandControllerMock.tasksInListForProject)
+                        .toBeCalledWith(taskList, projectId);
+                    done();
+                    return;
+                }))
+                .build();
+
+            client.emit('message', message);
+        });
+
+        it('should send warning when channel type is not text', () => {
+            const client = setupClient();
+
+            const commandControllerMock = new CommandControllerMockBuilder()
+                .withTasksInListForProject(jest.fn(() => Promise.resolve(new RichEmbed())))
+                .build();
+
+            const discordController = new DiscordControllerBuilder()
+                .withCommandController(commandControllerMock)
+                .withClient(client)
+                .build();
+
+            const message = new MessageBuilder()
+                .withContent(`!tasks in list`)
+                .withChannelType('voice')
+                .build();
+
+            client.emit('message', message);
+
+            expect(message.channel.send)
+                .toBeCalledWith('!tasks in list command must be called from' 
+                    + ' a text channel associated with a project');
+        });
+
+        it('should send warning and log error when error getting project ID', done => {
```

Comment:
Inconsistent naming - either change the code to log an error or rename the test to "log warning". Since it's caused by an unhandled exception, I would lean towards logging an error.

Reply:
Changed to log errors